spec/graphql/schema/object_spec.rb

```diff
@@ -324,6 +324,16 @@ def call(o, a, c)
       end
     end
 
+    it "warns when override matches field name" do
+      expected_warning = "X's `field :object` conflicts with a built-in method, use `resolver_method:` to pick a different resolver method for this field (for example, `resolver_method: :resolve_object` and `def resolve_object`)\n"
+      assert_output "", expected_warning do
+        Class.new(GraphQL::Schema::Object) do
+          graphql_name "X"
+          field :object, String, null: true, resolver_method: :object
```

Comment:
This use case is an interesting one, not quite sure what's best here but for now to keep the check in `has_fields` simpler instead of over complicating things I think this is best. Would love your thoughts?

Reply:
Not sure there's a 100% perfect solution so imo this is fine